components/speedreader/speedreader_util.cc

```diff
@@ -46,4 +51,32 @@ bool URLReadableHintExtractor::HasHints(const GURL& url) {
   return false;
 }
 
+void SetSiteSpeedreadable(HostContentSettingsMap* map,
+                          const GURL& url,
+                          bool enable) {
+  DCHECK(!url.is_empty());  // Not supported. Disable Speedreader in settings.
+
+  // Rule covers all protocols and pages.
+  auto pattern = ContentSettingsPattern::FromString("*://" + url.host() + "/*");
+  if (!pattern.IsValid())
+    return;
+
+  ContentSetting setting =
+      enable ? CONTENT_SETTING_BLOCK : CONTENT_SETTING_ALLOW;
+  map->SetContentSettingCustomScope(pattern, ContentSettingsPattern::Wildcard(),
+                                    ContentSettingsType::BRAVE_SPEEDREADER,
+                                    setting);
+}
+
+bool IsEnabledForURL(HostContentSettingsMap* map, const GURL& url) {
+  ContentSetting setting = map->GetContentSetting(
+      url, GURL(), ContentSettingsType::BRAVE_SPEEDREADER);
+  // Since Brave specific content settings are registered with
+  // CONTENT_SETTINCG_BLOCK, the logic is reversed here. Sites that are
```

Comment:
We shouldn't do this, it's extremely confusing. Register the default as allow in the content settings registry

Reply:
fixed